### DISCUSSION TITLE
Update Luarocks Lua dep from lua51 to lua54

### DISCUSSION THREAD

**Discussion by GR3YH4TT3R93** [termux/termux-packages#19252] [Ideas]
Would it be possible/ a good idea to update Luarocks' deps?

I'm not deep into the lua scene so I'm not sure if any issues would arise but from a quick look at the arch package, it's using the base lua package for arch which is currently the latest version of lua 5.4 (specifically, lua 5.4.6-3) so I would think it's at least worth looking into for up-to-date features/ not having two versions of lua installed.

**Comment by TomJo2000** [comment]
So the issue is that there is 3(-ish) commonly used, more or less compatible, major implementations of Lua.
- Lua 5.1
- Lua Latest (5.4.x at this point in time)
- LuaJIT (currently API compatible with Lua 5.1)

Unless specific Lua 5.2 through 5.4 features are used the standard target for most Lua scripts is going to be LuaJIT or 5.1.

I'm afraid you're likely not gonna get around needing multiple versions of Lua installed since each package on Luarocks can specify its own Lua version as a dependency.

For what its worth, Luarocks itself can make use of any version of Lua 5.x, though doesn't officially support building with LuaJIT.
([Nixpkgs seems to do it anyway](https://github.com/NixOS/nixpkgs/blob/release-23.11/pkgs/development/tools/misc/luarocks/default.nix#L45-L60) for their luarocks package.)
We could build luarocks against 5.4 as Arch does, but there's little to no benefit.
There is no functional difference in what version of Lua the Lua package manager is using.
